artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -3159,6 +3162,9 @@ public void run() {
                         connection.killMessage(server.getNodeID());
                         remotingService.removeConnection(connection.getID());
                         connection.fail(ActiveMQMessageBundle.BUNDLE.connectionsClosedByManagement(connection.getRemoteAddress()));
+                        //break once a consumer gets killed. This can prevent all
+                        //consumers to this queue get killed all at once.
+                        break;
```

Comment:
you will also need to change your test as you made an assumption a consumer would be killed. Since the overal rate doesn't satisfy the condition of the threshold no consumers should be killed..


@gaohoward I will let you handle these changes.